modin/experimental/core/io/sql/sql_dispatcher.py

```diff
@@ -0,0 +1,129 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Module houses `SQLExperimentalDispatcher` class."""
+
+import warnings
+
+import pandas
+import numpy as np
+
+from modin.core.io import SQLDispatcher
+from modin.config import NPartitions
+
+
+class SQLExperimentalDispatcher(SQLDispatcher):
+    """Class handles experimental utils for reading SQL queries or database tables."""
+
+    __read_sql_with_offset = None
+
+    @classmethod
+    def preprocess_func(cls):  # noqa: RT01
+        """Prepare a function for transmission to remote workers."""
+        if cls.__read_sql_with_offset is None:
+            from modin.experimental.core.io.sql.utils import read_sql_with_offset
+
+            # cls.__read_sql_with_offset = cls.put(read_sql_with_offset)
+            cls.__read_sql_with_offset = read_sql_with_offset
+        return cls.__read_sql_with_offset
+
+    @classmethod
+    def _read(
+        cls,
+        sql,
+        con,
+        index_col,
+        coerce_float,
+        params,
+        parse_dates,
+        columns,
+        chunksize,
+        partition_column,
+        lower_bound,
+        upper_bound,
+        max_sessions,
+    ):  # noqa: PR01
+        """
+        Read SQL query or database table into a DataFrame.
+
+        Documentation for parameters can be found at `modin.read_sql`.
+
+        Returns
+        -------
+        BaseQueryCompiler
+            A new query compiler with imported data for further processing.
+        """
+        from modin.experimental.core.io.sql.utils import (
+            is_distributed,
+            get_query_info,
+        )
+
+        if not is_distributed(partition_column, lower_bound, upper_bound):
+            warnings.warn("Defaulting to Modin core implementation")
```

Comment:
shouldn't this warning give a reason on why we default?

Reply:
I updated the message